src/Estina/SmartFocus/Api/Rest/Notification.php

```diff
@@ -98,6 +99,10 @@ public function buildTransactionalRequestObject(
         $enableTracking = false,
         array $additionalParams = null
     ) {
+        if (empty($additionalParams['uidkey'])) {
```

Comment:
A tiny note, I know you need to keep the order of parameters sent to this method, but adding this exception made all parameters mandatory, and no need to made them optional.
I think we cannot do anything about this.

Reply:
Yes we can't break the method signature. I don't think removing = null on $additionalParams would be helpful either because you'd get a generic type error instead of a message explaining which particular param you need to set.